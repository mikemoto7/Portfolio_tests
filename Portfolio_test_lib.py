
from test_flags import _product_called_from_test
_product_called_from_test = True


import sys
sys.path.append("../lib_for_test")

from general_test_lib import do_test_runstring
import general_test_lib

general_test_lib.product_dir_path = "../Portfolio"







