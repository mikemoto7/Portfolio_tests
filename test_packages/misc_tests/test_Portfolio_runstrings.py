import pytest

from Portfolio_test_lib import do_test_runstring


# @pytest.mark.usefixtures("oneTimeSetUp", "setUp")
@pytest.mark.usefixtures()
class TestPortfolio():

    # @pytest.fixture(autouse=True)
    # def classSetup(self, setup):
    #     pass



    # Testcases interactive:

    '''
    def test_runstring_cl(self):
        do_test_runstring("Portfolio.py --cl")

    def test_runstring_cl_l(self):
        do_test_runstring("Portfolio.py --cl l")

    def test_runstring_cl_7(self):
        do_test_runstring("Portfolio.py --cl 7")

    def test_runstring_cl_debug_1(self):
        do_test_runstring("Portfolio.py --cl 7 --debug 1")
    '''


    # Testcases from docstring:


    def test_runstring_price_web_update(self):
        do_test_runstring("Portfolio.py --price_web_update --pfile demo_data_file.xml")

    def test_runstring_ac_web_update(self):
        do_test_runstring("Portfolio.py --ac_web_update --pfile demo_data_file.xml")

    def test_runstring_stats(self):
        do_test_runstring("Portfolio.py --stats --pfile demo_data_file.xml")

    def test_runstring_xml(self):
        do_test_runstring("Portfolio.py --xml --pfile demo_data_file.xml")

    def test_runstring_price_web_update_xml(self):
        do_test_runstring("Portfolio.py --price_web_update --xml --pfile demo_data_file.xml")

    def test_runstring_pt_show(self):
        do_test_runstring("Portfolio.py --pt show --pfile demo_data_file.xml")

    def test_runstring_pt_show_percent(self):
        do_test_runstring("Portfolio.py --pt show_percent --pfile demo_data_file.xml")

    def test_runstring_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --pt show_ac_misc_totals --pfile demo_data_file.xml")

    def test_runstring_pt_lp(self):
        do_test_runstring("Portfolio.py --pt lp --pfile demo_data_file.xml")

    def test_runstring_pt_add(self):
        do_test_runstring("Portfolio.py --pt add:'Small Value',Non_Retirement,50000:'Small Value',Retirement,20000:Healthcare,Retirement,50000 --pfile demo_data_file.xml")

    def test_runstring_inv_show(self):
        do_test_runstring("Portfolio.py --inv show --pfile demo_data_file.xml")

    def test_runstring_inv_diff_1file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml")

    def test_runstring_inv_diff_2file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml1,demo_data_file.xml2")

    def test_runstring_inv_lots(self):
        do_test_runstring("Portfolio.py --inv lots --pfile demo_data_file.xml")

    def test_runstring_all_not_silent(self):
        do_test_runstring("Portfolio.py --all --pfile demo_data_file.xml")

    def test_runstring_all_silent(self):
        do_test_runstring("Portfolio.py --all --silent --pfile demo_data_file.xml")

    def test_runstring_noindent(self):
        do_test_runstring("Portfolio.py --noindent  --all  --pfile demo_data_file.xml")

    def test_runstring_h(self):
        do_test_runstring("Portfolio.py --h")

    def test_runstring_xmlfile(self):
        do_test_runstring("Portfolio.py --xmlfile  --pfile demo_data_file.xml")

    def test_runstring_pdt(self):
        do_test_runstring("Portfolio.py --pdt --pfile demo_data_file.xml")

    def test_runstring_debug_1(self):
        do_test_runstring("Portfolio.py --debug 1  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_2(self):
        do_test_runstring("Portfolio.py --debug 2  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_12(self):
        do_test_runstring("Portfolio.py --debug 12  --all  --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_show(self):
        do_test_runstring("Portfolio.py --price_web_update --inv show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_ac_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --ac_web_update --pt show%lp%add:Small_Value,Retirement,13000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_lp(self):
        do_test_runstring("Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Large_Blend,Retirement,13000%show_percent%lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_diff(self):
        do_test_runstring("Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent_bigger_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_ac_misc_totals --pfile demo_data_file.xml")






