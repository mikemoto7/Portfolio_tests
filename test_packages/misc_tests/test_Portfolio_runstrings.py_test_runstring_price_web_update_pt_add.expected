
Processing: user_runstring = ./Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml


Processing: --price_web_update

Processing: --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

-----------------------------------------------------------------------------------------------------------
                          Tot        Tot    Tot        Ret        Ret    Ret    Non_Ret    Non_Ret  Non_Ret
Asset_Class         inv_count        amt    pct  inv_count        amt    pct  inv_count        amt      pct
Energy                      4   97837.40  11.39          3   77939.40  15.04          1   19898.00     5.84
Global                      6   57632.60   6.71          4   48694.30   9.40          2    8938.30     2.62
Healthcare                  2   25640.10   2.98          1    4135.50   0.80          1   21504.60     6.31
Individual_Stocks           1   25878.60   3.01          0       0.00   0.00          1   25878.60     7.59
Large_Blend                 3   43508.60   5.06          2   23145.60   4.47          1   20363.00     5.97
Large_Growth                4  107245.50  12.48          2   77404.00  14.94          2   29841.50     8.75
Large_Value                 3   47526.30   5.53          2   32897.70   6.35          1   14628.60     4.29
Mid_Blend_Aggress           3   73973.90   8.61          0       0.00   0.00          3   73973.90    21.70
Mid_Growth                  5   85482.90   9.95          3   69038.00  13.32          2   16444.90     4.82
Mid_Value                   5   54082.80   6.30          3   30378.20   5.86          2   23704.60     6.95
REIT                        5  114252.00  13.30          4   79805.80  15.40          1   34446.20    10.10
Small_Blend                 2   23496.80   2.74          1    4477.20   0.86          1   19019.60     5.58
Small_Growth                4   82504.50   9.60          3   67261.80  12.98          1   15242.70     4.47
Small_Value                 2   20006.80   2.33          1    2970.00   0.57          1   17036.80     5.00
Totals                     49  859068.80  99.99         29  518147.50  99.99         20  340921.30    99.99
                                                                                                           

-----------------------------------------------------------------------------------------------------------
                          Tot        Tot    Tot        Ret        Ret    Ret    Non_Ret    Non_Ret  Non_Ret
Asset_Class         inv_count        amt    pct  inv_count        amt    pct  inv_count        amt      pct
Energy                      4   97837.40  11.39          3   77939.40  15.04          1   19898.00     5.84
Global                      6   57632.60   6.71          4   48694.30   9.40          2    8938.30     2.62
Healthcare                  2   25640.10   2.98          1    4135.50   0.80          1   21504.60     6.31
Individual_Stocks           1   25878.60   3.01          0       0.00   0.00          1   25878.60     7.59
Large_Blend                 3   43508.60   5.06          2   23145.60   4.47          1   20363.00     5.97
Large_Growth                4  107245.50  12.48          2   77404.00  14.94          2   29841.50     8.75
Large_Value                 3   47526.30   5.53          2   32897.70   6.35          1   14628.60     4.29
Mid_Blend_Aggress           3   73973.90   8.61          0       0.00   0.00          3   73973.90    21.70
Mid_Growth                  5   85482.90   9.95          3   69038.00  13.32          2   16444.90     4.82
Mid_Value                   5   54082.80   6.30          3   30378.20   5.86          2   23704.60     6.95
REIT                        5  114252.00  13.30          4   79805.80  15.40          1   34446.20    10.10
Small_Blend                 2   23496.80   2.74          1    4477.20   0.86          1   19019.60     5.58
Small_Growth                4   82504.50   9.60          3   67261.80  12.98          1   15242.70     4.47
Small_Value                 2   20006.80   2.33          1    2970.00   0.57          1   17036.80     5.00
Totals                     49  859068.80  99.99         29  518147.50  99.99         20  340921.30    99.99
                                                                                                           

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                    Before                        After                        Before                        After                        Before                        After
Asset                Tot_%    Asset               Tot_%    Asset                Ret_%    Asset               Ret_%    Asset                Non_%    Asset               Non_%
Small_Value           2.33    Small_Value          2.33    Mid_Blend_Aggress     0.00    Mid_Blend_Aggress    0.00    Global                2.62    Global               2.62
Small_Blend           2.74    Small_Blend          2.74    Small_Value           0.57    Small_Value          0.57    Large_Value           4.29    Large_Value          4.29
Healthcare            2.98    Healthcare           2.98    Healthcare            0.80    Healthcare           0.80    Small_Growth          4.47    Small_Growth         4.47
Large_Blend           5.06    Large_Blend          5.06    Small_Blend           0.86    Small_Blend          0.86    Mid_Growth            4.82    Mid_Growth           4.82
Large_Value           5.53    Large_Value          5.53    Large_Blend           4.47    Large_Blend          4.47    Small_Value           5.00    Small_Value          5.00
Mid_Value             6.30    Mid_Value            6.30    Mid_Value             5.86    Mid_Value            5.86    Small_Blend           5.58    Small_Blend          5.58
Global                6.71    Global               6.71    Large_Value           6.35    Large_Value          6.35    Energy                5.84    Energy               5.84
Mid_Blend_Aggress     8.61    Mid_Blend_Aggress    8.61    Global                9.40    Global               9.40    Large_Blend           5.97    Large_Blend          5.97
Small_Growth          9.60    Small_Growth         9.60    Small_Growth         12.98    Small_Growth        12.98    Healthcare            6.31    Healthcare           6.31
Mid_Growth            9.95    Mid_Growth           9.95    Mid_Growth           13.32    Mid_Growth          13.32    Mid_Value             6.95    Mid_Value            6.95
Energy               11.39    Energy              11.39    Large_Growth         14.94    Large_Growth        14.94    Large_Growth          8.75    Large_Growth         8.75
Large_Growth         12.48    Large_Growth        12.48    Energy               15.04    Energy              15.04    REIT                 10.10    REIT                10.10
REIT                 13.30    REIT                13.30    REIT                 15.40    REIT                15.40    Mid_Blend_Aggress    21.70    Mid_Blend_Aggress   21.70
Individual_Stocks     3.01    Individual_Stocks    3.01    Individual_Stocks     0.00    Individual_Stocks    0.00    Individual_Stocks     7.59    Individual_Stocks    7.59
Totals               99.99                        99.99                         99.99                        99.99                         99.99                        99.99

